# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/IonBuilder.h
# Commit: 2c25a18171ba
# Full Hash: 2c25a18171ba0a181781d6395b2e6443e26ac30e
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-11-26 16:16:58
# Regressor Bug: 1599117
# File Overlap Count: 1
# Description:
#   Bug 1599117 part 2 - Restore empty-block + improveTypesForTest behavior for test instructions. r=tcampbell
#   
#   When an MTest instruction results in a triangle structure, we now create an
#   empty block (like we did before bug 1595476) to get a diamond structure.
#   We did this for short-circuit operators but not for other cases.
# ==============================================================================

diff -r 95a4904b7465 -r 2c25a18171ba js/src/jit/IonBuilder.h
--- a/js/src/jit/IonBuilder.h	Tue Nov 26 11:27:18 2019 +0000
+++ b/js/src/jit/IonBuilder.h	Tue Nov 26 11:27:59 2019 +0000
@@ -354,6 +354,8 @@
   // Improve the type information at tests
   AbortReasonOr<Ok> improveTypesAtTest(MDefinition* ins, bool trueBranch,
                                        MTest* test);
+  AbortReasonOr<Ok> improveTypesAtTestSuccessor(MTest* test,
+                                                MBasicBlock* successor);
   AbortReasonOr<Ok> improveTypesAtCompare(MCompare* ins, bool trueBranch,
                                           MTest* test);
   AbortReasonOr<Ok> improveTypesAtNullOrUndefinedCompare(MCompare* ins,
